modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/data/set/filter/AutocompleteClayDataSetFilterContextContributor.java

```diff
@@ -41,20 +41,21 @@
 
 		if (clayDataSetFilter instanceof BaseAutocompleteClayDataSetFilter) {
 			return _serialize(
-				(BaseAutocompleteClayDataSetFilter)clayDataSetFilter);
+				(BaseAutocompleteClayDataSetFilter)clayDataSetFilter, locale);
 		}
 
 		return Collections.emptyMap();
 	}
 
 	private Map<String, Object> _serialize(
-		BaseAutocompleteClayDataSetFilter baseAutocompleteClayDataSetFilter) {
+		BaseAutocompleteClayDataSetFilter baseAutocompleteClayDataSetFilter,
+		Locale locale) {
 
 		return HashMapBuilder.<String, Object>put(
 			"apiURL", baseAutocompleteClayDataSetFilter.getAPIURL()
 		).put(
 			"inputPlaceholder",
-			baseAutocompleteClayDataSetFilter.getPlaceholder()
+			baseAutocompleteClayDataSetFilter.getPlaceholder(locale)
```

Comment:
After discussing it in Slack we have come to the conclusion that moving the localization to the `AutocompleteClayDataSetFilterContextContributor` is the best way to avoid the breaking change.

That way we don't have to break `BaseAutocompleteClayDataSetFilter`'s contract with `AutocompleteClayDataSetFilterContextContributor`, because the contract is that `BaseAutocompleteClayDataSetFilters` return keys and `AutocompleteClayDataSetFilterContextContributor` takes care of the localization.

It even feels cleaner, I think, as it is better to localize strings as late as possible.